crates/agglayer-telemetry/src/clock.rs

```diff
@@ -0,0 +1,144 @@
+//! Clock metrics for observability
+//!
+//! This module provides comprehensive metrics for monitoring the Agglayer block
+//! clock, including connection status, block processing, epoch tracking, and
+//! health monitoring.
+
+use lazy_static::lazy_static;
+use opentelemetry::{global, metrics::*};
+
+const AGGLAYER_CLOCK_OTEL_SCOPE_NAME: &str = "agglayer_node_clock";
+
+lazy_static! {
+    /// Current block height gauge
+    pub static ref CURRENT_BLOCK_HEIGHT: Gauge<u64> = global::meter(AGGLAYER_CLOCK_OTEL_SCOPE_NAME)
+        .u64_gauge("current_block_height")
+        .with_description("Current block height processed by the clock")
+        .build();
+
+    /// Current epoch number gauge
+    pub static ref CURRENT_EPOCH: Gauge<u64> = global::meter(AGGLAYER_CLOCK_OTEL_SCOPE_NAME)
+        .u64_gauge("current_epoch")
+        .with_description("Current epoch number")
+        .build();
+
+    /// Gauge for connection status (1 = connected, 0 = disconnected)
+    pub static ref CONNECTION_STATUS: Gauge<u64> = global::meter(AGGLAYER_CLOCK_OTEL_SCOPE_NAME)
+        .u64_gauge("connection_status")
+        .with_description("WebSocket connection status (1=connected, 0=disconnected)")
+        .build();
+
+    /// Counter for reconnection attempts
+    pub static ref RECONNECTION_ATTEMPTS: Counter<u64> = global::meter(AGGLAYER_CLOCK_OTEL_SCOPE_NAME)
+        .u64_counter("reconnection_attempts_total")
+        .with_description("Total number of reconnection attempts")
+        .build();
+
+    /// Gauge for health status (1=healthy, 0.5=starting, 0.25=degraded, 0=unhealthy)
+    pub static ref HEALTH_STATUS: Gauge<f64> = global::meter(AGGLAYER_CLOCK_OTEL_SCOPE_NAME)
+        .f64_gauge("health_status")
+        .with_description("Current health status of the block clock")
+        .build();
+
+    /// Gauge for seconds since last block received (for health monitoring)
+    pub static ref SECONDS_SINCE_LAST_BLOCK: Gauge<u64> = global::meter(AGGLAYER_CLOCK_OTEL_SCOPE_NAME)
+        .u64_gauge("seconds_since_last_block")
+        .with_description("Seconds since the last block was received")
+        .build();
+
+    /// Gauge for seconds since last epoch event (for health monitoring)
+    pub static ref SECONDS_SINCE_LAST_EPOCH: Gauge<u64> = global::meter(AGGLAYER_CLOCK_OTEL_SCOPE_NAME)
+        .u64_gauge("seconds_since_last_epoch")
+        .with_description("Seconds since the last epoch event was sent")
+        .build();
+
+    /// Counter for subscription lag events
+    pub static ref SUBSCRIPTION_LAG: Counter<u64> = global::meter(AGGLAYER_CLOCK_OTEL_SCOPE_NAME)
+        .u64_counter("subscription_lag_total")
+        .with_description("Total number of subscription lag events")
+        .build();
```

Comment:
Maybe mention in the name this refers to lag to subscription to *blocks*? Just "subscription" can refer to any number of things.